### PR TITLE
removed outdated CI instructions

### PR DESCRIPTION
**Pull request by varianceslayer**
Removed outdated CI instructions as mentioned in issue #10522

### CODE REVIEW

File: sphinx/source/docs/dev_guide/testing.rst [hunk 1]
```diff
@@ -215,21 +215,14 @@ There are a number of files that affect the build configuration:
 Etiquette
 ~~~~~~~~~
 
-CI services provide finite free build workers to Open Source projects. A few
-considerations will help you be considerate of others needing these limited
+CI services provide finite free build workers to Open Source projects. The below
+consideration will help you be considerate of others needing these limited
 resources:
 
 * Group commits into meaningful chunks of work before pushing to GitHub (i.e.
```
Comment (tcmetzger) [inline sphinx/source/docs/dev_guide/testing.rst:222]: Removing the second item makes this a one-item bullet list. I would recommend rewriting this into one cohesive paragraph and not have a bullet list at all.

**Review by tcmetzger — COMMENTED** [review]
Thank you for your help with Bokeh's documentation!

**Comment by varianceslayer** [comment]
Made the changes as requested and added a paragraph instead of a bullet list.

**Review by tcmetzger — APPROVED** [review]
Thanks!

**Comment by bryevdv** [comment]
job failed from conda network error, restarted

**Review by pavithraes — APPROVED** [review]

**Comment by pavithraes** [comment]
@varianceslayer Thank you!

**Pull request merged**